### PR TITLE
Add Snapshot support

### PR DESCRIPTION
**Pull request by pierre-emmanuelJ**
`exo snapshot`
image

### Create a snapshot with a name or id of instance

`exo snapshot create vmTest`
image

### you can delete one snapshot or more at the same time

`exo snapshot delete rt_ROOT-690946_20181108115920`
image

### You can list all snapshot or list by instance name or id

`exo snapshot ls`
image

`exo snapshot ls rt`
image

### or give multiple instrances 

`exo snapshot ls vm1 vm2 vm3`

### you can revert one or more snapshot at the same time

`exo snapshot revert fbda5929-9752-443c-a2d2-386ec2ab3403 `
image

### CODE REVIEW

File: cmd/snapshot.go [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+
+import (
+	"github.com/exoscale/egoscale"
+	"github.com/spf13/cobra"
+)
+
+// snapshotCmd represents the snapshot command
+var snapshotCmd = &cobra.Command{
+	Use:   "snapshot",
+	Short: "Snapshots allow you to save the volume of your machine in it's current state",
```
Comment (pyr) [inline cmd/snapshot.go]:
```suggestion
	Short: "Snapshots allow you to save the volume of a machine in its current state",
```

File: cmd/snapshot_create.go [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+		volume := &egoscale.Volume{
+			VirtualMachineID: vm.ID,
+			Type:             "ROOT",
+		}
+
+		if err := cs.GetWithContext(gContext, volume); err != nil {
+			return err
+		}
+
+		res, err := asyncRequest(&egoscale.CreateSnapshot{VolumeID: volume.ID}, fmt.Sprintf("Create Snapshot of %q", vm.Name))
```
Comment (pyr) [inline cmd/snapshot_create.go]:
```suggestion
		res, err := asyncRequest(&egoscale.CreateSnapshot{VolumeID: volume.ID}, fmt.Sprintf("Creating snapshot of %q", vm.Name))
```

File: cmd/snapshot_delete.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+		}
+
+		tasks := make([]task, 0, len(args))
+
+		for _, arg := range args {
+			volume, err := getSnapshotWithNameOrID(arg)
+			if err != nil {
+				return err
+			}
+			t := task{egoscale.DeleteSnapshot{ID: volume.ID}, fmt.Sprintf("Delete Snapshot %q", volume.Name)}
```
Comment (pyr) [inline cmd/snapshot_delete.go]:
```suggestion
			t := task{egoscale.DeleteSnapshot{ID: volume.ID}, fmt.Sprintf("Deleting snapshot %q", volume.Name)}
```

File: cmd/snapshot_list.go [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+var snapshotListCmd = &cobra.Command{
+	Use:     "list [vm name | vm id]+",
+	Short:   "List snapshot",
+	Aliases: gListAlias,
+	RunE: func(cmd *cobra.Command, args []string) error {
+
+		table := table.NewTable(os.Stdout)
+
+		if len(args) == 0 {
+			table.SetHeader([]string{"State", "Create On", "Size", "ID"})
```
Comment (pyr) [inline cmd/snapshot_list.go]:
```suggestion
			table.SetHeader([]string{"State", "Created On", "Size", "ID"})
```

File: cmd/snapshot_list.go [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+				snapshot := s.(*egoscale.Snapshot)
+				table.Append([]string{snapshot.State, snapshot.Created, fmt.Sprintf("%v", humanize.IBytes(uint64(snapshot.Size))), snapshot.ID.String()})
+			}
+
+			table.Render()
+
+			return nil
+		}
+
+		table.SetHeader([]string{"VM", "State", "Create On", "Size", "ID"})
```
Comment (pyr) [inline cmd/snapshot_list.go]:
```suggestion
		table.SetHeader([]string{"VM", "State", "Created On", "Size", "ID"})
```

File: cmd/snapshot_revert.go [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+			return cmd.Usage()
+		}
+
+		tasks := make([]task, 0, len(args))
+		for _, arg := range args {
+			snapshot, err := getSnapshotWithNameOrID(arg)
+			if err != nil {
+				return err
+			}
+			task := task{egoscale.RevertSnapshot{ID: snapshot.ID}, fmt.Sprintf("Revert snapshot %q", snapshot.Name)}
```
Comment (pyr) [inline cmd/snapshot_revert.go]:
```suggestion
			task := task{egoscale.RevertSnapshot{ID: snapshot.ID}, fmt.Sprintf("Reverting snapshot %q", snapshot.Name)}
```

**Comment by pierre-emmanuelJ** [comment]
@pyr thanks 👍

**Comment by greut** [comment]
delete should ask for a confirmation as well.

**Comment by SebastienPittet** [comment]
I just upgraded to exo version 1.0.4 (from 0.11.6) . Many thanks for snapshot support. Very appreciated! It's efficient, working fine !

**Comment by pierre-emmanuelJ** [comment]
Thank you for your feedback :smile:

**Pull request merged**